embassy-usb/src/lib.rs

```diff
@@ -165,6 +165,25 @@ struct Interface {
     num_alt_settings: u8,
 }
 
+/// A report of the used size of the runtime allocated buffers
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct UsbBufferReport {
+    /// Number of device descriptor bytes used
+    pub device_descriptor_used: usize,
+    /// Number of config descriptor bytes used
+    pub config_descriptor_used: usize,
+    /// Number of bos descriptor bytes used
+    pub bos_descriptor_used: usize,
+    /// Number of msos descriptor bytes used
+    ///
+    /// Will be `None` if the "msos-descriptor" feature is not active.
+    /// Otherwise will return Some(bytes).
+    pub msos_descriptor_used: Option<usize>,
```

Comment:
I decided to go with an option here to differentiable between "msos available but not used" and "msos not active", rather than feature flagging (since this is not on a hot path/persistent usage or anything). I figured that would make printing and such less tedious for folks.

Let me know if you'd prefer a feature flag'd usize instead.

Reply:
lgtm. I think we should remove that feature anyway, or at least not make it change the `Builder::new()` signatur, because that's quite annoying.